pkg/starboard/config.go

```diff
@@ -332,7 +332,21 @@ func (c *configManager) EnsureDefault(ctx context.Context) error {
 		Data: GetDefaultConfig(),
 	}
 	_, err := c.client.CoreV1().ConfigMaps(c.namespace).Create(ctx, cm, metav1.CreateOptions{})
-	if !apierrors.IsAlreadyExists(err) {
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: c.namespace,
+			Name:      SecretName,
+			Labels: labels.Set{
+				"app.kubernetes.io/managed-by": "starboard",
+			},
+		},
+	}
```

Comment:
It looks like the secret is initialized with empty data. Where are values filled? A user edits it manually?

Reply:
Yup. It is supposed to be UPDATED or PATCHED manually with kubectl (Starboard CLI) or programmatically with Kubernetes API (Aqua Wave)

Reply:
For example, a user needs to set `trivy.githubToken`, right?
https://github.com/aquasecurity/starboard/blob/bbd497b0df6755c6c05cd286bbafd89e93a24ff6/pkg/trivy/scanner.go#L128

Does Starboard CLI support to set it? Or, does a user configure it via raw kubectl?

Reply:
The starboard.ConfigManager is used by Starboard CLI and the operator. And yes, users are supposed to edit the secret with `kubectl` until we implement `starboard config set -from-literal=key1=config1`. Currently we just have a command to read config and hide implementation details `starboard config get --name-only`.

The configuration steps are documented in https://github.com/aquasecurity/starboard#configuration
I will update it in another PR while migrating from README to MkDocs, but the only difference is that we introduced secret to configure sensitive params such as `trivy.githubToken`.

Beyond that, there will be more values that we should not store in ConfigMap such as Trivy server tokens, Aqua CSP credentials, etc.

Reply:
I didn't know the document.
https://github.com/aquasecurity/starboard#configuration

It makes sense. Thanks.